src/gba/vfame.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright (c) 2016 taizou
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Support for copy protected unlicensed games from the company Vast Fame
+
+#ifndef GBA_VFAME_H
+#define GBA_VFAME_H
+
+#include "util/common.h"
+
+enum GBAVFameCartType {
+	VFAME_NO = 0,
+	VFAME_STANDARD = 1,
+	VFAME_GEORGE = 2
+};
+
+struct GBAVFameCart {
+	enum GBAVFameCartType cartType;
+	int sramMode;
+	int romMode;
+	int8_t writeSequence[5];
+	char acceptingModeChange;
+};
```

Comment:
ah yeah, good point about save states, I didn't think about that.  For the most part save states should work without these being saved, at least as long as the ROM mirroring stuff is disabled; the games always seem to change SRAM mode immediately before doing a write, so the state of these modes beforehand doesn't usually matter. But definitely something I'll look at later.